File: articles/institutional-path-dependence/go/path_dependence_score.go
package main

import "fmt"

type PathDependenceCase struct {
	InitialConditions       float64
	BehavioralReinforcement float64
	FeedbackStrength        float64
	IncreasingReturns       float64
	CoordinationEffects     float64
	LearningEffects         float64
	Legitimacy              float64
	SwitchingCosts          float64
	Complementarity         float64
	DisruptionPressure      float64
	ReformCapacity          float64
}

func PathDependenceScoreRaw(x PathDependenceCase) float64 {
	return 0.08*x.InitialConditions +
		0.12*x.BehavioralReinforcement +
		0.12*x.FeedbackStrength +
		0.13*x.IncreasingReturns +
		0.11*x.CoordinationEffects +
		0.10*x.LearningEffects +
		0.12*x.Legitimacy +
		0.12*x.SwitchingCosts +
		0.10*x.Complementarity -
		0.12*x.DisruptionPressure -
		0.05*x.ReformCapacity
}

func main() {
	demo := PathDependenceCase{80, 75, 70, 82, 78, 74, 85, 88, 79, 35, 40}
	fmt.Printf("Institutional path dependence raw score: %.2f\n", PathDependenceScoreRaw(demo))
}

File: articles/institutional-path-dependence/rust/path_dependence_score.rs
struct PathDependenceCase {
    initial_conditions: f64,
    behavioral_reinforcement: f64,
    feedback_strength: f64,
    increasing_returns: f64,
    coordination_effects: f64,
    learning_effects: f64,
    legitimacy: f64,
    switching_costs: f64,
    complementarity: f64,
    disruption_pressure: f64,
    reform_capacity: f64,
}

fn path_dependence_score_raw(x: &PathDependenceCase) -> f64 {
    0.08 * x.initial_conditions
        + 0.12 * x.behavioral_reinforcement
        + 0.12 * x.feedback_strength
        + 0.13 * x.increasing_returns
        + 0.11 * x.coordination_effects
        + 0.10 * x.learning_effects
        + 0.12 * x.legitimacy
        + 0.12 * x.switching_costs
        + 0.10 * x.complementarity
        - 0.12 * x.disruption_pressure
        - 0.05 * x.reform_capacity
}

fn main() {
    let demo = PathDependenceCase {
        initial_conditions: 80.0,
        behavioral_reinforcement: 75.0,
        feedback_strength: 70.0,
        increasing_returns: 82.0,
        coordination_effects: 78.0,
        learning_effects: 74.0,
        legitimacy: 85.0,
        switching_costs: 88.0,
        complementarity: 79.0,
        disruption_pressure: 35.0,
        reform_capacity: 40.0,
    };

    println!(
        "Institutional path dependence raw score: {:.2}",
        path_dependence_score_raw(&demo)
    );
}
